docs/plugins.md

```diff
@@ -85,64 +85,224 @@ the output of Protractor by returning a results object.
 Plugins are node modules which export an object with the following API:
 
 ```js
-/*
+/**
  * Sets up plugins before tests are run. This is called after the WebDriver
  * session has been started, but before the test framework has been set up.
  *
- * @param {Object} config The plugin configuration object. Note that
- *     this is not the entire Protractor config object, just the
- *     entry in the plugins array for this plugin.
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
  *
- * @return Object If an object is returned, it is merged with the Protractor
- *     result object. May return a promise.
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
  */
-exports.setup = function(config) {};
+exports.setup = function() {};
 
-/*
+/**
  * This is called after the tests have been run, but before the WebDriver
  * session has been terminated.
  *
- * @param {Object} config The plugin configuration object.
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
  *
- * @return Object If an object is returned, it is merged with the Protractor
- *     result object. May return a promise.
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
  */
-exports.teardown = function(config) {};
+exports.teardown = function() {};
 
-/*
+/**
  * Called after the test results have been finalized and any jobs have been
  * updated (if applicable).
  *
- * @param {Object} config The plugin configuration object.
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, it is outputted to the console
  *
- * @return Return values are ignored.
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
```

Comment:
Good catch.  However, protractor really will wait on a promise if it's returned.  And if the promise is rejected, an error will be logged to the console now